docs/topics/arrays.md

```diff
@@ -31,32 +60,293 @@ fun main() {
-have the same set of methods and properties. Each of them also has a corresponding factory function:
+To randomly shuffle the elements in an array, use the [shuffle()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/shuffle.html) function.
+
+```Kotlin
+fun main() {
+// sampleStart
+    val simpleArray = arrayOf(1, 2, 3)
+
+    // Shuffles elements [3, 2, 1]
+    simpleArray.shuffle()
+    println(simpleArray.joinToString())
+
+    // Shuffles elements again [2, 3, 1]
+    simpleArray.shuffle()
+    println(simpleArray.joinToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="shuffle-array-kotlin"}
+
+### Zip
+
+To combine two arrays into an array of [`Pair`]((https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/)), use the [zip()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/zip.html) function.
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val firstArray = arrayOf(1, 2, 3)
+    val secondArray = arrayOf(1, 2, 3)
+
+    // Zips arrays [(1, 1), (2, 2), (3, 3)]
+    val zipArray = firstArray.zip(secondArray)
+    println(zipArray.joinToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="zip-array-kotlin"}
+
+To return a pair of lists from an array of `Pair`, use the [unzip()](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/unzip.html) function.
+
+```Kotlin
+fun main() { 
+//sampleStart
+    val pairArray = arrayOf(Pair(1,1), Pair(2,2), Pair(3,3))
+
+    // Unzips array ([1, 2, 3], [1, 2, 3])
+    val unzipArray = pairArray.unzip()
+    println(unzipArray)
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="unzip-array-kotlin"}
+
+## Concatenate arrays
+
+Arrays of type `String` can be concatenated using the `+` operator.
 
 ```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf("a", "b", "c")
+    val anotherArray = arrayOf("d", "e", "f")
+    
+    // Creates a new array by concatenating elements: ["a", "b", "c", "d", "e", "f"]
+    println((simpleArray + anotherArray).joinToString())
+//sampleEnd
+}
 ```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="concatenate-array-kotlin"}
+
+## Use arrays with variable number of arguments
+
+In some cases it is useful to be able to pass a variable number of arguments to a function without having to define the number of arguments in advance.
+In Kotlin you can use the `vararg` parameter for this. If you want to pass your variable number of arguments to a function as the contents of an array, use the _spread_ operator (`*`).
+The spread operator passes each element of your array as individual arguments to your chosen function.
+
+```kotlin
+fun main() {
+    val lettersArray = arrayOf("c", "d")
+    val lettersList = listOf("e", "f")
+    printAllStrings("a", "b", *lettersArray, *lettersList.toTypedArray())
```

Comment:
If you want to keep the `lettersList` in here, it would be required to explain briefly what the `toTypedArray()` function does. Actually, `toTypedArray()` seems like a valuable addition to this page (Something along the lines of "Converting collections to arrays")
Also required to document: the difference between `toTypedArray` and `toIntArray` / the other primitives. Those can be related back to the sections between primitive arrays and object-arrays as well.